### PR TITLE
Update README

### PR DESCRIPTION
**Pull request by mgaeta**
Make the README more readable and useful.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,45 +1,76 @@
...
+The following are all _required_.
 
-## Troubleshooting
-Building and running this action locally on an unsupported environment yields an error that looks like:
-```
-Syntax error: end of file unexpected (expecting ")")
-```
+|Name|Description|
+|---|---|
+|`SENTRY_AUTH_TOKEN`|See [authentication documentation](https://docs.sentry.io/api/auth). |
```
Comment (jonesphillip) [inline README.md:39]: I don't think we want to be pointing people down this path. This points people to generate auth tokens associated with users. We should point people to create internal integrations.

Example: https://github.com/getsentry/sentry-netlify-build-plugin#create-a-sentry-internal-integration

File: README.md [hunk 2]
```diff
@@ -1,45 +1,76 @@
...
+`production` environment of your project with an automatically generated `version` name.
   
 ```yaml
-uses: getsentry/action-release@v1
-with:
-  environment: 'production'
-  sourcemaps: './lib'
+jobs:
+  release:
+    runs-on: ubuntu-latest
```
Comment (jonesphillip) [inline README.md:12]: Do you think it makes sense to have the outer definition here?

File: README.md [hunk 3]
```diff
@@ -1,45 +1,76 @@
...
+    env:
+      SENTRY_AUTH_TOKEN: ${{ secrets.SENTRY_AUTH_TOKEN }}
+      SENTRY_ORG: ${{ secrets.SENTRY_ORG }}
+      SENTRY_PROJECT: ${{ secrets.SENTRY_PROJECT }}
+    steps:
+    - uses: actions/checkout@v2
+    - uses: getsentry/action-release@v1.0.0
```
Comment (jonesphillip) [inline README.md:19]: Can can we add the name parameter for our action?

Also do we need to define the environment in the top level? Can we bundle this under the Sentry step?

File: README.md [hunk 4]
```diff
@@ -1,45 +1,76 @@
...
-```
-Syntax error: end of file unexpected (expecting ")")
-```
+|Name|Description|
+|---|---|
+|`SENTRY_AUTH_TOKEN`|See [authentication documentation](https://docs.sentry.io/api/auth). |
+|`SENTRY_ORG`|Your organization slug on your Sentry instance.|
+|`SENTRY_PROJECT`|The name of your project on Sentry.|
+
+### Examples
```
Comment (jonesphillip) [inline README.md:43]: @manuzope do we think it'd be valuable to include a full workflow example here? Show users where this fits, in specific how to trigger this on push or on release?

File: README.md [hunk 5]
```diff
@@ -1,45 +1,76 @@
...
-## Why Releases?
-A release is a version of your code that is deployed to an environment. 
-When you give Sentry information about your releases, you unlock a number of new features:
- - Determine the issues and regressions introduced in a new release
- - Predict which commit caused an issue and who is likely responsible
- - Resolve issues by including the issue number in your commit message
- - Receive email notifications when your code gets deployed
-
-Additionally, releases are used for applying source maps to minified JavaScript to view original, untransformed source code.
+Automatically create a Sentry release in a workflow.
```
Comment (jonesphillip) [inline README.md:2]: Think we should explain why creating releases, deploys, sending commit metadata to Sentry is valuable

File: README.md (hunk 4 above)
Comment (manuzope) [reply README.md:43]: Yes. Full example would be useful :+1:

File: README.md (hunk 1 above)
Comment (mgaeta) [reply README.md:39]: This is a really good point. For now I'll copy in what we have for Netlify.

File: README.md (hunk 2 above)
Comment (mgaeta) [reply README.md:12]: I wanted to capture the environment variables, but it'll be a lot simpler if I pass them into the step.

File: README.md (hunk 3 above)
Comment (mgaeta) [reply README.md:19]: 👍

**Pull request merged**